cpp/simulations/abm.cpp

```diff
@@ -22,20 +22,35 @@
 #include "abm/household.h"
 #include "abm/state.h"
 #include "memilio/io/result_io.h"
+#include "memilio/utils/uncertain_value.h"
 #include "boost/filesystem.hpp"
 
 namespace fs = boost::filesystem;
 
+/**
+ * Set a value and distribution of an UncertainValue.
+ * Assigns average of min and max as a value and UNIFORM(min, max) as a distribution.
+ * @param p uncertain value to set.
+ * @param min minimum of distribution.
+ * @param max minimum of distribution.
+ */
+void assign_uniform_distribution(mio::UncertainValue& p, double min, double max)
+{
+    p = mio::UncertainValue(0.5 * (max + min));
+    p.set_distribution(mio::ParameterDistributionUniform(min, max));
+}
+
 /**
  * Determine the infection state of a person at the beginning of the simulation.
  * The infection states are chosen randomly. They are distributed according to the probabilites set in the example.
  * @return random infection state
  */
-mio::abm::InfectionState determine_infection_state(double exposed, double infected, double carrier, double recovered)
+mio::abm::InfectionState determine_infection_state(mio::UncertainValue exposed, mio::UncertainValue infected,
+                                                   mio::UncertainValue carrier, mio::UncertainValue recovered)
 {
-    double susceptible          = 1 - exposed - infected - carrier - recovered;
-    std::vector<double> weights = {susceptible,  exposed,      carrier,       infected / 3,
-                                   infected / 3, infected / 3, recovered / 2, recovered / 2};
+    mio::UncertainValue susceptible = 1 - exposed - infected - carrier - recovered;
+    std::vector<ScalarType> weights = {susceptible,  exposed,      carrier,       infected / 3,
+                                       infected / 3, infected / 3, recovered / 2, recovered / 2};
```

Comment:
I don't see the point in changing it to UncertainValues here as the function only performs simple arithmetic on some values.

Reply:
It is fixed!